### ISSUE TITLE
How to deal with intermediate ranks not used in backbone?

### ISSUE THREAD

**Issue by CecSve** [gbif/portal-feedback#4504]
On helpdesk (January 2nd 2023), a correspondence was initiated on whether GBIF should be more strict with ranks and provide a much higher penalty for wrong rank "groups", i.e. suprafamily, family, genus, species & infraspecific names which will be continued in this issue.  

There is a similar problems with other higher taxa which are also often rare genera, e.g. Vertebrata. Its a simple change, but might have more impact than desired, e-g- if a very wrong rank was given. The coding change would be simple if we think it's worth trying.

@mdoering can you expand on how such a penalty would work?

**Comment by mdoering** [comment]
All matching is based on an overall matching score (confidence) that is the summary of various individual matching scores. If that goes below a threshold, I believe its 80 currently, the match is discarded as a NO_MATCH.

You can see the individual scores in the remarks when verbose is on:

https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata
> Similarity: name=100; authorship=0; classification=-2; rank=0; status=1; singleMatch=5

https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=phylum
> Similarity: name=100; authorship=0; classification=-2; rank=-35; status=1; singleMatch=5

Individual scores can be negative, e.g. the above plant genus Vertebrata is not considered a match anymore when the rank phylum is given. Rank similarity scores are set to 6 if equal, otherwise the distance in the rank enumeration is calculated and used as a negative score (penalty). As phylum and genus is quite a bit apart that leads to -35 now which makes a match hardly possible already. So really in those cases we already have a big penalty which prevents matches. The problem we had were the unknown rank `superdomain` which result in a zero rank score: https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=superdomain

If we added superdomain to the ranks enum this would address the problem.
We could also score all unknown (=`other`) ranks badly, but that would lead to more no matches for good name matches when there is an akward rank given, e.g. in some non english language we dont yet support. The German term [Gattung](https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=gattung) is understood, but the [dutch Geslacht](https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=geslacht) and likely many others not. So I think it is too dangerous to provide a penalty to all unknown ranks.

The original idea with rank groups was that there are distinct groups of ranks (infraspecific/trinomial, species/binomial, genus, family and above groups) so we could increase the penalty if the match is between such groups and increase the score slightly if its still the same group. species, genus & family group names are explicitly dealt with in the zoological code and you can just use a genus name as a subgenus without publishing it as a new name. 

Simply adding superdomain to our rank enum would probably adress our problem at hand best though.

**Comment by CecSve** [comment]
> Individual scores can be negative, e.g. the above plant genus Vertebrata is not considered a match anymore when the rank phylum is given. Rank similarity scores are set to 6 if equal, otherwise the distance in the rank enumeration is calculated and used as a negative score (penalty). As phylum and genus is quite a bit apart that leads to -35 now which makes a match hardly possible already. So really in those cases we already have a big penalty which prevents matches. The problem we had were the unknown rank `superdomain` which result in a zero rank score: https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=superdomain

Thank you for explaining this in detail and I see your point about that it might muddy the picture if unknown ranks would receive a negative score. 

I suspect it is not the enum resource you are referring to, but I can see the superdomain -> domain mapping has been added to [this resource](https://github.com/gbif/parsers/blob/master/src/main/resources/dictionaries/parse/rank.tsv). Is it this resource you mean?

**Comment by mdoering** [comment]
Indeed wIth that rank parser it should work, but we don't seem to be using it in the deployed version. Will need to update

**Comment by CecSve** [comment]
There's also this resource: https://github.com/gbif/gbif-api/blob/dev/src/main/java/org/gbif/api/vocabulary/Rank.java

**Comment by mdoering** [comment]
Yes, thats the GBIF rank enum. The name parser maps its own, richer enum to the GBIF one.
Ideally we'd just have a single enum in name parser...

**Comment by CecSve** [comment]
Just to be sure - will a fix be applied now @mdoering and I can close the issue?

**Comment by mdoering** [comment]
we could deploy a new version with that fix if its desired. Shall we do a release? It will only apply to new interpretations, but we could reinterpret the few problem cases we know about

**Comment by mdoering** [comment]
@CecSve I have released a new version and deployed it to UAT where it behaves as expected:
https://api.gbif-uat.org/v1/species/match?verbose=true&name=Vertebrata&rank=superdomain

I will deploy this to prod early next week

**Comment by timrobertson100** [comment]
[This returns](https://api.gbif.org/v1/species/match?verbose=true&name=Vertebrata&rank=superdomain) `No match because of too little confidence`. Is that expected please @mdoering?

**Comment by mdoering** [comment]
Yes, thats the expected result. And without a match it would be linked to incertae sedis

**Comment by mdoering** [comment]
Same for Biota: https://api.gbif.org/v1/species/match?verbose=true&name=Biota&rank=Superdomain

**Comment by CecSve** [comment]
Thank you - I am looking into which datasets are affected and could be fixed by the new version. It is unfortunately a bit more complex than what I anticipated and I will make a new issue to track the process. For example, missing taxonRanks ([example](https://www.gbif.org/occurrence/datasets?taxon_key=7326344&advanced=1&network_key=2b7c7b4f-4d4f-40d3-94de-c28b6fa054a6&occurrence_status=present)), [using BiotaNA instead of Biota in scientificName](https://www.gbif.org/occurrence/datasets?advanced=1&verbatim_scientific_name=BiotaNA&occurrence_status=present) (issue created for the dataset with most occurrences https://github.com/gbif/portal-feedback/issues/4534)

FYI, the fix with assigning taxonRank = kingdom also seems to [work](https://www.gbif.org/occurrence/search?dataset_key=69217c7b-5773-4015-a802-6af216b24c97&advanced=1&verbatim_scientific_name=Biota), which we ideally would like to have as the same end result for taxonRank = superdomain.

**Comment by mdoering** [comment]
@timrobertson100 Species match cache maybe? We did not clear anything, just deploy a new nub-ws service

**Comment by timrobertson100** [comment]
Thanks @mdoering - I follow now.

I've flushed the caches, reprocessed [this record ](https://api.gbif.org/v1/occurrence/3742897314) which should be calling [this lookup](https://api.gbif.org/v1/species/match?verbose=true&name=Biota&rank=Superdomain) which returns as you expect.

I'll explore what the pipeline is doing on this

**Comment by timrobertson100** [comment]
Moving this to pipelines
